### PR TITLE
Validation for CIBA binding_message parameter

### LINKED ISSUE #9469: Validation for CIBA binding_message parameter
### Description

The CIBA specification mentions this about `binding_message` parameter in this section https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.7.1 :
```
Because the various devices involved may have limited display abilities and the message is intending for visual inspection by 
the end-user, the binding_message value SHOULD be relatively short and use a limited set of plain text characters. The 
invalid_binding_message defined in Section 13 is used in the case that it is necessary to inform the Client that the provided 
binding_message is unacceptable. 
```

Keycloak currently allows any binding message, but it may be better to limit this according to the specification. For example some other FAPI CIBA certified providers (Authlete, Raidiam) has limitation on the binding message to be limited in length and have only plain-text characters like this regex: `^[a-zA-Z0-9-._+/!?#]{1,50]$ `

Moreover, changing it this way will help to have FAPI CIBA automated conformance testsuite working (without restrictions on the binding message, the automated test `fapi-ciba-id1-ensure-authorization-request-with-potentially-bad-binding-message` is currently failing in the automated testsuite and testsuite always need to be run manually to pass)

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mposolda**
closes #9469

### CODE REVIEW

**Comment by mposolda** [comment]
@tnorimat Do you please have a chance to review this? See #9469 for the details.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/CIBATest.java [hunk 1]
```diff
@@ -397,6 +398,39 @@ public void testLoginHintTokenRequiredButNotSend() throws Exception {
...
+            clientRep = clientResource.toRepresentation();
+            prepareCIBASettings(clientResource, clientRep);
+
+            // Binding message with non plain-text characters
+            String bindingMessage = "1234 \uD83D\uDC4D\uD83C\uDFFF 品川 Lor";
+            AuthenticationRequestAcknowledgement response = oauth.doBackchannelAuthenticationRequest(TEST_CLIENT_NAME, TEST_CLIENT_PASSWORD, username, bindingMessage, null);
+            assertThat(response.getStatusCode(), is(equalTo(400)));
+            assertThat(response.getError(), is(OAuthErrorException.INVALID_BINDING_MESSAGE));
+
+            // Long binding message
```
Comment (tnorimat) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/CIBATest.java:418]: It is not necessary but how about adding test for vacant binding message like
```
        bindingMessage ="";
```

**Review by tnorimat — DISMISSED** [review]
@mposolda I've reviewed it and add one comment. Could you check it?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/CIBATest.java (hunk 1 above)
Comment (mposolda) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/CIBATest.java:418]: Good point, I've added the test for this scenario. Could you please re-review?

**Review by tnorimat — APPROVED** [review]
@mposolda LGTM

**Comment by mposolda** [comment]
@tnorimat Thanks!

**Pull request merged**